politeiad/api/v1/v1.go

```diff
@@ -40,6 +40,7 @@ const (
 	ChallengeSize      = 32         // Size of challenge token in bytes
 	TokenSize          = 32         // Size of token
 	MetadataStreamsMax = uint64(16) // Maximum number of metadata streams
+	TokenPrefixLength  = 8          // Length of prefix of token used for lookups
```

Comment:
Why did you pick a length of 8?  What's the math behind it?

Reply:
I picked 8 because it's relatively short, but it still gives us over 4 billion choices, so it will still be easy to find unique tokens when there are over 1 billion records.

Reply:
I don't think we need to worry about politeia handling a billion records.  We can scale that down a couple orders of magnitude.  If we're going to shorten the token for a UX improvement we might as well shorten it more than that.

The token is hex so a prefix length of 6 would mean there is a `1/(16^6)` chance of any two prefixes colliding.  So even if you have 100,000 records, which pi really can't support in its current form, there would be a 0.6% chance of a prefix collision.  That's fine imo.  The token generation function you made retries up to 1000 times.  No way will that max out.

This token prefix is not actually being saved as part of the git repo, it's just a row in the cache, so if it needs to be scaled up in the future it's as simple as changing the TokenPrefixLength, rebuilding the cache, and updating the route.

Reply:
Yeah that makes sense. For the proposal system and cms, 6 is definitley good. I was mostly thinking about keeping this large enough so it wouldn't limit any other projects that may want to use politeiad as a back end, but I guess they could just fork the code and update this if they need to.